lib/models/file/middleware/remove-associated.js

```diff
@@ -1,6 +1,14 @@
 async function middleware (file) {
   const FileAssignment = require('../../file-assignment');
   await FileAssignment.find({ file: file._id }).cursor().eachAsync(record => record.remove());
+
+  if (file.correctiveAction) {
+    const correctiveAction = require('../../corrective-action');
+
+    await correctiveAction.update(
+      { documents: file.id },
```

Comment:
`file.id` isn't a thing
```suggestion
      { documents: file._id },
```